omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,179 @@
+	}
+
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertNotEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+	
+	@Test
+	public void shouldSaveAllergyWithCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+        Assert.assertEquals(allergies.size(), 4);
+
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"35d3346a-6769-4d52-823f-b4b234bac3e3\"} }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);		
+        Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+        Assert.assertEquals(allergies.size(), 4);
+		
+		// this is the proper sequence to set nonCodedAllergen
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, \"nonCodedAllergen\": \"test non coded\" }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("test non coded", Util.getByPath(newAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+        Assert.assertEquals(allergies.size(), 5);
+	}
+
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldSetNoKnownAllergiesWithNonExistingPatient() throws Exception
+	{
+		String nonExistentPatientURI = "patient/nonExistingPatient/allergy";

```

Comment:
The method name does not match with what you are testing. In other words, what are you trying to test in this method?